### PR TITLE
Add explicit casting to double (instead of float)

### PR DESCRIPTION
**Pull request by carlopi**

### CODE REVIEW

File: tests/unit/codegen/test12.cpp [hunk 1]
```diff
@@ -10,5 +10,5 @@ void webMain()
 {
 	const char* str="-0.832050323486328125";
 	double ret=strtod(str, NULL);
-	assertEqual(ret,-0.832050323486328125,"High precision strtod");
+	assertEqual<double>(ret,(double)-0.832050323486328125,"High precision strtod");
```
Comment (alexp-sssup) [inline tests/unit/codegen/test12.cpp:13]: Literals are double by default. What is the problem supposed to be?

Comment (carlopi) [reply tests/unit/codegen/test12.cpp:13]: I will check again, but basically the problem was that the float specialization of assertEqual was taken:
```
template<class T>
void assertEqual(const T value, const T expected, const char* msg)
{
        assertEqualImpl(value==expected, msg);
}
```

Comment (alexp-sssup) [reply tests/unit/codegen/test12.cpp:13]: It cannot make sense, at the very least `ret` is certainly double. So T would be double. But anyway the literal is certain double as well.

**Comment by carlopi** [comment]
Not sure exactly what I saw, at the IR level the float version was taken, but I am not able to reproduce (as in now the IR shows correctly the double version is taken).
Closing this, good catch.

**Pull request closed without merging**